opentelemetry-sdk/src/logs/simple_log_processor.rs

```diff
@@ -60,54 +59,47 @@ use std::sync::Mutex;
 ///
 #[derive(Debug)]
 pub struct SimpleLogProcessor<T: LogExporter> {
-    exporter: Mutex<T>,
-    is_shutdown: AtomicBool,
+    exporter: T,
+    is_exporting: AtomicBool,
 }
 
 impl<T: LogExporter> SimpleLogProcessor<T> {
     /// Creates a new instance of `SimpleLogProcessor`.
     pub fn new(exporter: T) -> Self {
         SimpleLogProcessor {
-            exporter: Mutex::new(exporter),
-            is_shutdown: AtomicBool::new(false),
+            exporter,
+            is_exporting: AtomicBool::new(false),
         }
     }
 }
 
 impl<T: LogExporter> LogProcessor for SimpleLogProcessor<T> {
     fn emit(&self, record: &mut SdkLogRecord, instrumentation: &InstrumentationScope) {
-        // noop after shutdown
-        if self.is_shutdown.load(std::sync::atomic::Ordering::Relaxed) {
-            // this is a warning, as the user is trying to log after the processor has been shutdown
-            otel_warn!(
-                name: "SimpleLogProcessor.Emit.ProcessorShutdown",
-            );
-            return;
+        // export() does not require mutable self and can be called in parallel
+        // with other export() calls. However, OTel Spec requires that
+        // existing export() must be completed before the next export() call.
+        while !self
+            .is_exporting
+            .compare_exchange(false, true, Ordering::Acquire, Ordering::Relaxed)
```

Comment:
This looks inefficient to me - spin lock with busy loop. I understand the processor is not for production but still don't see if we really need to add this. If Mutex is not really the way, should we use the mpsc channel instead.

Reply:
Yes, I can use mutex and block until lock is acquired. (All I wanted is to *not* wrap Exporter in mutex.)

Reply:
this is done!